### PR TITLE
[FLINK-27739] Add benchmark module for table store

### PR DESCRIPTION
**Pull request by tsreaper**
Currently there is no benchmark for table store to measure its performance. We need such module.

### CODE REVIEW

**Review by JingsongLi — COMMENTED** [review]
Thank @tsreaper for the contribution. I just have a question, can we run this benchmark in local machine?
Can it also make some sense if it's just a single parallelism?

File: flink-table-store-benchmark/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  * One master node with 8 cores and 16GB RAM.
+  * Two worker nodes with 8 cores and 32GB RAM.
+* This benchmark runs on a standalone Flink cluster. Download Flink >= 1.15 from the [Apache Flink's website](https://flink.apache.org/downloads.html#apache-flink-1150) and setup a standalone cluster. Flink's job manager must be on the master node of your EMR cluster. We recommend the following Flink configurations:
+    ```yaml
+    jobmanager.memory.process.size: 8192m
+    taskmanager.memory.process.size: 4096m
+    taskmanager.numberOfTaskSlots: 1
+    parallelism.default: 16
+    execution.checkpointing.interval: 3min
+    state.backend: rocksdb
```
[the remark this answers is not in the document]
Comment (tsreaper) [reply flink-table-store-benchmark/README.md:17]: Yes. `CommitterOperator` in table store has state.

**Comment by tsreaper** [comment]
> Thank @tsreaper for the contribution. I just have a question, can we run this benchmark in local machine? Can it also make some sense if it's just a single parallelism?

This benchmark can run on local machines. I, however, suspect if running on local machines makes any sense. The same with running with just a single parallelism.

File: flink-table-store-benchmark/src/main/java/org/apache/flink/table/store/benchmark/metric/BenchmarkMetric.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import javax.annotation.Nullable;
+
+import java.util.Objects;
+
+/**
+ * Metric collected per {@link
+ * org.apache.flink.table.store.benchmark.BenchmarkOptions#METRIC_MONITOR_DURATION} for a single
+ * query.
+ */
+public class BenchmarkMetric {
```
Comment (JingsongLi) [inline flink-table-store-benchmark/src/main/java/org/apache/flink/table/store/benchmark/metric/BenchmarkMetric.java:32]: Should we add `copied from nexmark` for classes?

File: flink-table-store-benchmark/pom.xml [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    <name>Flink Table Store : Benchmark</name>
+
+    <properties>
+        <flink.table.api.java.bridge.name>flink-table-api-java-bridge</flink.table.api.java.bridge.name>
+        <flink.streaming.java.name>flink-streaming-java</flink.streaming.java.name>
+    </properties>
+
+    <profiles>
+        <profile>
+            <id>flink-1.14</id>
```
Comment (JingsongLi) [inline flink-table-store-benchmark/pom.xml:41]: we can use the profile in parent pom.

File: flink-table-store-benchmark/src/main/java/org/apache/flink/table/store/benchmark/metric/BenchmarkMetric.java (hunk 2 above)
Comment (tsreaper) [reply flink-table-store-benchmark/src/main/java/org/apache/flink/table/store/benchmark/metric/BenchmarkMetric.java:32]: No need. I've asked @wuchong and he says that we just need to add `inspired by Nexmark` in `README.md`.

**Review by JingsongLi — APPROVED** [review]
Looks good to me!

**Pull request merged**